### PR TITLE
Add a Mini-spec for `cursorTextColor`

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

The issue here is that we draw the cursor _on top_ of the text. This spec introduces a new setting that lets us:
* draw the cursor on top of the text
* draw the cursor underneath the text
* draw the cursor underneath the text, _and_ provide a specific color for the text drawn on top of the cursor.
* draw the cursor underneath the text, _and_ use the text background color as the foreground for the text drawn on top of the cursor.

## PR Checklist
* [x] This is a spec for #1203
* [x] I work here

## Detailed Description of the Pull Request / Additional comments
_read the spec_

### CODE REVIEW

**Comment by DHowett** [comment]
Concerns from a chat on Friday:

* This will break ligatures under the cursor
   * If we explicitly redraw the cell under the cursor, we may draw half a ligature plus a normal character after it :(
* Because the background is drawn with the run, drawing the cursor _before_ a run with a background will result in the cursor being destroyed

**Comment by DHowett** [comment]
Filed https://github.com/microsoft/terminal/issues/6193 as a discussion platform for one way to fix cursor-behind-text.

File: doc/specs/#1203 - Cursor Text Color.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+that the cell under the text cursor is always totally obscured when the cursor
+is in "filledBox" mode.
+
+This spec outlines the addition of a new setting `cursorTextColor`, that would
+allow the text under the cursor to be drawn in another color, so that the text
+will always be visible.
+
+### Note
+
+This spec is being writtin a bit retroactively. [#6337] was merged before this
```
Comment (zadjii-msft) [inline doc/specs/#1203 - Cursor Text Color.md:32]:
```suggestion
This spec is being written a bit retroactively. [#6337] was merged before this
```

**Comment by DHowett** [comment]
#7118 is the followup for lighting up invert and perhaps others.

**Comment by zadjii-msft** [comment]
This has gone fairly stale, and doesn't address the concerns about ligatures. I think #6193 is the direction we want to go with this _eventually_. So I'm just gonna close this out for now.

**Pull request closed without merging**